lib/engine-native/src/artifact.rs

```diff
@@ -317,7 +317,7 @@ impl NativeArtifact {
         if is_cross_compiling {
             Self::from_parts_crosscompiled(metadata, shared_filepath)
         } else {
-            let lib = Library::new(&shared_filepath).map_err(to_compile_error)?;
+            let lib = unsafe { Library::new(&shared_filepath).map_err(to_compile_error)? };
```

Comment:
Technically this is part of a different PR?

Reply:
Yeah but I want to see green tests in this PR for the moment.